django-backend/fecfiler/contacts/views.py

```diff
@@ -162,13 +162,16 @@ def candidate_lookup(self, request):
         if office:
             params["office"] = office
         params = urlencode(params)
-        response = requests.get(
-            FEC_API_CANDIDATE_LOOKUP_ENDPOINT, params=params
-        )
-        if response.status_code != status.HTTP_404_NOT_FOUND:
-            response.raise_for_status()
+        if len(q) >= 3:
+            response = requests.get(
+                FEC_API_CANDIDATE_LOOKUP_ENDPOINT, params=params
+            )
+            if response.status_code != status.HTTP_404_NOT_FOUND:
+                response.raise_for_status()
 
-        json_results = response.json()
+            json_results = response.json()
+        else:
+            json_results = {}
```

Comment:
These were the source of the Contact Lookup bug from FECFILE-2323.  Essentially, if the query was less than 3 characters long, then the FEC API would respond with a 422 error.  `raise_for_status()` would then throw an error, which would interrupt the rest of the function.  This prevented our API from returning _any_ contacts on Committee and Candidate lookup if the search query was not at least 3 characters long.

Reply:
(Another valid approach here would have been to add the 422 error to the status codes ignored, but I figured it would be a good idea to avoid throwing requests at the FEC API that we already know will fail)